components/neeva/renderer/content_filtering_agent.cc

```diff
@@ -154,8 +154,8 @@ ContentFilteringPolicy ContentFilteringAgent::GetPolicyForRequest(
     // There's no point in blocking foo.com from making requests from first-party origins (e.g. foo.com). 
     // Doing so can break logins or other site functionality. 
     if (!is_url_third_party) {
-      // Downgrade from kBlockRequests to kBlockCookies to minimize impact on sites. 
-      return ContentFilteringPolicy::kBlockCookies;
+      // Downgrade from kBlockRequests or kBlockCookies to kAllow minimize impact on first-party-origin sites.
+      return ContentFilteringPolicy::kAllow;
```

Comment:
At this point, it would be a reasonable optimization to move this check up to line 136 because there's no point in calling `FindMatch` for these URLs. That function is not free, so might as well not call it when you are going to ignore the result.